httpclient/src/test/kotlin/tbdex/sdk/httpclient/E2ETest.kt

```diff
@@ -0,0 +1,233 @@
+        paymentDetails = mapOf(
+          "accountNumber" to "1234567890",
+          "reason" to "gift",
+          "accountHolderName" to "lainey wilson"
+        )
+      ),
+      claims = listOf(vcJwt)
+    )
+    val rfq = Rfq.create(pfiDid, myDid.uri, rfqData)
+    rfq.sign(myDid)
+    println("Sending RFQ against first offering id: ${offerings[0].metadata.id}")
+    val sendRfqResponse = client.sendMessage(rfq)
+
+    if (sendRfqResponse is ErrorResponse) {
+      throw AssertionError(
+        "Error returned from sending RFQ. " +
+          "Errors: ${sendRfqResponse.errors?.joinToString(", ") { it.detail }}"
+      )
+    }
+    println("SendRfqResponse is success?? ${sendRfqResponse is SendMessageResponse}")
+    println("Pinging for quote")
+
+    var listOfExchanges: List<List<Message>>
+    var currentExchange: List<Message>
+    var attempt = 0
+    do {
+      val getExchangeResponse: TbdexResponse = client.getExchanges(
+        pfiDid,
+        myDid,
+        GetExchangesFilter(listOf(rfq.metadata.exchangeId.toString()))
+      )
+
+      if (getExchangeResponse is ErrorResponse) {
+        throw AssertionError(
+          "Error returned from getting Exchanges. " +
+            "Errors: ${getExchangeResponse.errors?.joinToString(", ") { it.detail }}"
+        )
+      }
+
+      listOfExchanges = (getExchangeResponse as GetExchangesResponse).data
+      if (listOfExchanges.isEmpty()) {
+        throw AssertionError(
+          "No Exchange available " +
+            "(RFQ we sent earlier not being persisted!)"
+        )
+      }
+      println(
+        "Filtering list of exchanges for an exchange " +
+          "with exchangeId of ${rfq.metadata.exchangeId}"
+      )
+      currentExchange =
+        listOfExchanges.first { exchanges ->
+          exchanges.any { msg -> msg.metadata.exchangeId == rfq.metadata.exchangeId }
+        }
+      if (currentExchange.size < 2) {
+        Thread.sleep(pollInterval)
+        attempt++
+        println("Attempt #$attempt at fetching exchanges after sending RFQ")
+      }
+      if (attempt > 5) {
+        break
+      }
+    } while (currentExchange.size < 2)
+
+    if (currentExchange.last() !is Quote) {
+      throw AssertionError("Timed out trying to get a Quote")
+    }
+    val quote = currentExchange.last() as Quote
+    println("Got quote! QuoteData: Hurry, quote expires at ${quote.data.expiresAt}")
+
+    val order = Order.create(pfiDid, myDid.uri, rfq.metadata.exchangeId)
+    order.sign(myDid)
+
+    println("Sending order against Quote with exchangeId of ${order.metadata.exchangeId}")
+    val sendOrderResponse = client.sendMessage(order)
+
+    if (sendOrderResponse is ErrorResponse) {
+      throw AssertionError(
+        "Error returned from sending Order. " +
+          "Errors: ${sendOrderResponse.errors?.joinToString(", ") { it.detail }}"
+      )
+    }
+
+    println("Successfully sent order?? ${sendOrderResponse is SendMessageResponse}")
```

Comment:
Expect no order statuses if send order failed?

Reply:
yea same dealio here, i'm bailing if SendOrderResponse is ErrorResponse

```if (sendOrderResponse is ErrorResponse) {
      throw AssertionError(
        "Error returned from sending Order. " +
          "Errors: ${sendOrderResponse.errors?.joinToString(", ") { it.detail }}"
      )
    }``` removed these `am i success??` logs to mitigate confusion :)